### DISCUSSION TITLE
only able to connect to a wifi once with a usb dongle

### DISCUSSION THREAD

**Discussion by ktmtreck** [RaspAP/raspap-webgui#1587] [Q&A]
I have a problem connecting my raspberry pi 4B with a usb wifi dongle to a already existing wifi.

**Edit: I am using RaspAP docker**

I am using the pi's onboard wifi chip "wlan0" to generate an ap, where an arduino esp32 is exchanging mqtt messages with the pi.
Now I have purchased a fenix AX1800 wifi dongle to connect myself to the outside world.
After plugging it in the first time, I was able to connect to the desired wifi with the new wlan1 interface, so everything seemed to be alright.
After rebooting the pi, I can no longer scan for networks and I can only see the the network I was already connected with, but connecting again won't work.

If I want to rescan, nothing happens. If I want to delete the known network, I get the error message "Wifi settings updated but cannot restart (cannot execute "wpa_cli reconfigure")".
After searching for this problem, I found the FAQ post regarding the error (https://docs.raspap.com/faq/#wifi-scanning-doesnt-work-or-i-get-the-error-cannot-execute-wpa_cli-reconfigure)and followed the steps provided.

1. executing "sudo wpa_cli -i wlan0 scan_results or sudo wpa_cli -i wlan1 scan_results", it shows me results(each network multiple times) instead of the displayed error message of the faq.

2. not knowing what to do next, I proceeded with "sudo cat /etc/wpa_supplicant/wpa_supplicant.conf", where I ge this as a result:
ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
update_config=1

3. then I went on to reinitialize the socket, recieving the following console outputs for wlan0

```
sudo wpa_supplicant -B -Dnl80211,wext -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0
Successfully initialized wpa_supplicant
nl80211: kernel reports: Match already configured
nl80211: kernel reports: Match already configured
ctrl_iface exists and seems to be in use - cannot override it
Delete '/var/run/wpa_supplicant/wlan0' manually if it is not used anymore
Failed to initialize control interface 'DIR=/var/run/wpa_supplicant GROUP=netdev'.
You may have another wpa_supplicant process already running or the file was
left by an unclean termination of wpa_supplicant in which case you will need
to manually remove this file before starting wpa_supplicant again.

wlan0: CTRL-EVENT-DSCP-POLICY clear_all
wlan0: CTRL-EVENT-DSCP-POLICY clear_all
nl80211: deinit ifname=wlan0 disabled_11b_rates=0

```

and wlan1
 ```
sudo wpa_supplicant -B -Dnl80211,wext -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan1
Successfully initialized wpa_supplicant
nl80211: kernel reports: Match already configured
nl80211: kernel reports: Match already configured
```

4. after recieving this console output 
```
ctrl_iface exists and seems to be in use - cannot override it
Delete '/var/run/wpa_supplicant/wlan0' manually if it is not used anymore
Failed to initialize control interface 'DIR=/var/run/wpa_supplicant GROUP=netdev'.
You may have another wpa_supplicant process already running or the file was
left by an unclean termination of wpa_supplicant in which case you will need
to manually remove this file before starting wpa_supplicant again.
```

I decided to delete both the respective wlan0 and wlan1 files, and re ran the initialization command with the following output
```
sudo wpa_supplicant -B -Dnl80211,wext -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0
Successfully initialized wpa_supplicant
nl80211: kernel reports: Match already configured
Failed to create interface p2p-dev-wlan0: -16 (Device or resource busy)
nl80211: Failed to create a P2P Device interface p2p-dev-wlan0
P2P: Failed to enable P2P Device interface
```

and 
```
 sudo wpa_supplicant -B -Dnl80211,wext -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan1
Successfully initialized wpa_supplicant
nl80211: kernel reports: Match already configured
```

but still this does not seem to resolve the problem.
I still can not connect to the wifi and scanning for networks is still not working.
any help would be greatly appreciated.

bz edit: removed redundant log output

**Comment by billz** [comment]
Please refrain from pasting blocks of redundant log messages. This only makes parsing the discussion more difficult.

There have been _many_ conversations around this. Type `/` and enter: wpa_cli.
